docs/using-the-compiler.rst

```diff
@@ -44,6 +44,48 @@ If ``solc`` is called with the option ``--link``, all input files are interprete
 
 If ``solc`` is called with the option ``--standard-json``, it will expect a JSON input (as explained below) on the standard input, and return a JSON output on the standard output.
 
+.. _compiler-version:
+.. index:: ! compiler version, compile version
+
+Setting the compiler version
+******************************
+
+When you compile your contract code you can specify the compiler version to target or avoid particular features or behaviours in the release.
+
+.. warning::
+
+   Compiling for the wrong EVM version can result in wrong, strange and failing
+   behaviour. Please ensure, especially if running a private chain, that you
+   use the correct EVM version.
+
+You use the ``--evm-version`` option on the command line:
+
+.. code-block:: shell
+
+  solc   --evm-version <VERSION> contract.sol
+
+Or if using :ref:`JSON description format <compiler-api>`, with the ``evmVersion`` key:
+
+.. code-block:: json
+
+  {
+    "evmVersion": "<VERSION>"
+  }
+
+Compiler target options
+-----------------------
+
+- homestead
+- tangerineWhistle
+- spuriousDragon
+- byzantium (**default**)
+- constantinople
+
+.. note::
+
+  Backward compatibility might exist between different EVM versions, but it is
```

Comment:
This sounds rather passive-aggressively ;)

For a list of changes:

Starting from tangerineWhistle:
 - gas cost for access to other accounts increased (relevant for gas estimation and the optimizer)
 - for external calls, all gas is sent by default, previously, some delicate calculation had to be performed to not send too much

Starting from spuriousDragon:
 - gas cost for the exp opcode increased  (relevant for gas estimation and the optimizer)

Starting from byzantium:
 - opcodes `returndatacopy`, `returndatasize` and `staticcall` are available in assembly
 - the staticcall opcode is used when calling `view` or `pure` functions, which prevents them from modifying state at the EVM level (i.e. even applies when invalid type conversions are used)
 - it is possible to access dynamic data returned from function calls
 - TODO I think there was more - when was revert introduced?

Starting from constantinople (note that constantinople is not finalized yet and may change any time):
 - opcodes `shl`, `shr` and `sar` are available in assembly
 - shifting opcodes are used for shifting operators

Reply:
@chriseth This was paraphrased from @axic :) I initially thought that more detail about the versions was a better idea, but I wonder, are these changes mentioned elsewhere? It might be better just to link to that list.

Reply:
Since @chriseth collected such a complete list above I suggest to include in the documentation in this PR.

Reply:
It is a nice list, will do. @axic do you know the answer to the above?

> TODO I think there was more - when was revert introduced?

Reply:
`revert` was byzantium